### ISSUE TITLE
STG forgets pinned tabs, plus - when restarting it does not start with recent group

### ISSUE THREAD

**Issue by kpion** [Drive4ik/simple-tab-groups#588]
**Describe the bug**
For example I have a few tabs, two of them **pinned**. Now, when I restart FX, it starts with one page (which, btw, is the one set as 'default page for new tabs') and then, when I ask STG to open the group I had the last time, it does it, but without my two pinned tabs.

Now the weird thing - I tested it on a new profile, and it works fine. So, I got back to my default profile, disabled **all** the extensions and... still not working o.O

I'm really curious what is the reason, because I know I can kinda 'recreate' my default profile, but I started liking STG (I installed yesterday for the first time), and my guess is that I'll have this issue again the other day.

If I can do anything else (like turning on some logging or doing some more tests), then I'll be more than glad to do it.

**To Reproduce**
I have no idea :)

**Expected behavior**
Bringing back the last group I had active and restoring all the tabs, including pinned ones.

**Desktop (please complete the following information):**
 - OS and version: Kubuntu 18.04
 - Firefox version: 75
 - Simple Tab Groups version: 4.5

**Comment by C-Meinert** [comment]
If I remember correctly Pinned tabs are "special" and thus have some rules around what can be done with them.  For example STG cannot add pinned tabs to groups nor can they be hidden.

I would just validate that "Restore previous session" setting is enabled first, then perhaps check for conflicts with any other installed extensions.

**Comment by Drive4ik** [comment]
Yes, I would check for conflicts with other addons.
STG does not work with pinned tabs, does not move or affect them.
Only once when you create a backup and then restore, you can choose to restore pinned tabs, and that’s all.
The addon simply skips pinned tabs when loading a group, maybe another addon affects them.

**Comment by kpion** [comment]
@Drive4ik @C-Meinert 
> Check for conflicts with any other installed extensions.
> Yes, I would check for conflicts with other addons.

Thanks for the comments! 

Well, yeah, I temporarily disabled all my other extensions, and still not working, that is why it's so weird to me, because STG **does** work on a new, clear profile. It looks like my current profile has "something", but it's not an extension.

OK, well, I can live with that, I just stopped using pinned tabs :) So I'll let myself close this issue, so it won't fly around unnecessarily, it seems like I'm the only one having this issue. 

And, @Drive4ik, thanks for the cool extension!

**Comment by Drive4ik** [comment]
@kpion Try to peep on the `about:config` page
filter by `pinned`

**Comment by kpion** [comment]
@Drive4ik - thanks, but no, there is no difference there (withing those options having 'pinned' in the name) and a fresh new profile. 

But really, don't waste your time on this, please - apparently I was the only one with this issue, now even I don't have it :) Because I just stopped using pinned tabs :)

**Comment by Drive4ik** [comment]
> Because I just stopped using pinned tabs :)

This is not a solution to the problem))
I can advise moving bookmarks, passwords, tabs (STG backup) and more to a new profile and forget about this problem)

**Comment by kpion** [comment]
Hey, @Drive4ik - it's me again. Sorry for opening this old issue, I'll close it in a second.

It took me 8 months to find the reason of this behavior :) Just kidding, today I just tried again to solve it.

Well, it's actually pretty simple. Tabs are stored in a "Window Session" **\***, this also applies to pinned tabs. So, when I have multiple FF windows (with the same profile) opened, and then close them with the 'x' button on a window's title bar, FF remembers the tabs of the window which was closed as the last one (including pinned tabs). 

Your extension does remember "normal" tabs, but not the pinned tabs, AFAIK it's not possible for you to change that (I know it from another issue).

So the fact that this issue used to happen to me "randomly" was because sometimes I had only one FF window opened, and sometimes two or more. 

The solution: instead of using the 'x' button (which closes only this one window), we can use the 'quit' command (either from the menu bar (file->quit) or the hamburger menu (quit)). This has a drawback in that all of the FF windows (with this specific profile) are closed, but the good thing is that when we reopen FF, it restores everything like it was before closing - i.e. the tabs (including pinned ones) and this also solves the other thing I mentioned - FF also reopens the recently used group(s).

---
**\*** https://support.mozilla.org/en-US/questions/1291164
